### PR TITLE
BUG: Raise TypeError for scalar indexer in take

### LINKED ISSUE #42875: BUG: wrong/confusing error message in DataFrame.take with a scalar indexer
```
In [4]: df = pd.DataFrame({'a': [1, 2, 3]})

In [5]: df.take(1)
...

~/scipy/pandas/pandas/core/indexes/range.py in take(self, indices, axis, allow_fill, fill_value, **kwargs)
    433     ) -> Int64Index:
    434         with rewrite_exception("Int64Index", type(self).__name__):
--> 435             return self._int64index.take(
    436                 indices,
    437                 axis=axis,

~/scipy/pandas/pandas/core/indexes/base.py in take(self, indices, axis, allow_fill, fill_value, **kwargs)
    962             self._values, indices, allow_fill=allow_fill, fill_value=self._na_value
    963         )
--> 964         return type(self)._simple_new(taken, name=self.name)
    965 
    966     @final

~/scipy/pandas/pandas/core/indexes/base.py in _simple_new(cls, values, name)
    611         Must be careful not to recurse.
    612         """
--> 613         assert isinstance(values, np.ndarray), type(values)
    614 
    615         result = object.__new__(cls)

AssertionError: <class 'numpy.int64'>
```

`take` requires an array-like for the indexer, but when passing a scalar you get a not very useful error message as above. We should raise a ValueError (or rather TypeError) instead with an informative message.

It actually comes from the Index take implementation:

```
In [9]: pd.Index([1, 2, 3]).take(1)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-9-ce9f87ba8bd2> in <module>
----> 1 pd.Index([1, 2, 3]).take(1)

~/scipy/pandas/pandas/core/indexes/base.py in take(self, indices, axis, allow_fill, fill_value, **kwargs)
    962             self._values, indices, allow_fill=allow_fill, fill_value=self._na_value
    963         )
--> 964         return type(self)._simple_new(taken, name=self.name)
    965 
    966     @final

~/scipy/pandas/pandas/core/indexes/base.py in _simple_new(cls, values, name)
    611         Must be careful not to recurse.
    612         """
--> 613         assert isinstance(values, np.ndarray), type(values)
    614 
    615         result = object.__new__(cls)

AssertionError: <class 'numpy.int64'>
```

### PR DESCRIPTION
**Pull request by gfkang**
- [x] closes #42875
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1059,6 +1059,8 @@ def take(
     ):
         if kwargs:
             nv.validate_take((), kwargs)
+        if np.isscalar(indices):
+            raise TypeError("Expected indices to be array-like")
```
Comment (phofl) [inline pandas/core/indexes/base.py:1063]: Please use is_scalar from pandas.

File: pandas/tests/indexes/test_indexing.py [hunk 2]
```diff
@@ -70,6 +70,13 @@ def test_take(self, index):
             with pytest.raises(AttributeError, match=msg):
                 index.freq
 
+    def test_take_indexer_type(self):
+        integer_index = Int64Index([0, 1, 2, 3])
```
Comment (phofl) [inline pandas/tests/indexes/test_indexing.py:74]: I think this index constructor is deprecated, please use Index(

File: pandas/tests/indexes/test_indexing.py [hunk 3]
```diff
@@ -70,6 +70,13 @@ def test_take(self, index):
             with pytest.raises(AttributeError, match=msg):
                 index.freq
 
+    def test_take_indexer_type(self):
```
Comment (phofl) [inline pandas/tests/indexes/test_indexing.py:73]: Please add issue number

**Review by phofl — CHANGES_REQUESTED** [review]
Couple of comments. Also please add whatsnew

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Could you rebase master? We had a few ci failures

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @gfkang

**Pull request merged**